### PR TITLE
TEET-1058 Unique metadata

### PR DESCRIPTION
**Pull request by tatut**
Disallow editing a file if the operation would cause two files to have the same metadata.

### CODE REVIEW

File: app/backend/src/clj/teet/file/file_tx.clj [hunk 1]
```diff
@@ -67,11 +67,34 @@
...
+                        (d/q '[:find ?f
+                               :where
+                               [?t :task/files ?f]
+                               [(missing? $ ?f :meta/deleted?)]
+                               :in $ ?t]
+                             db task-id))
+        old-file (d/pull db modify-file-keys id)
         new-file (-> file
                      (select-keys modify-file-keys)
                      (update :file/part :db/id)
```
Comment (outamaa) [inline app/backend/src/clj/teet/file/file_tx.clj:84]: Not part of PR, but what's going on here? Collapsing map containing db id into just the db id itself. Why?

File: app/backend/src/clj/teet/file/file_tx.clj [hunk 2]
```diff
@@ -67,11 +67,34 @@
...
-                     cu/without-nils)]
-    (du/modify-entity-tx old-file new-file)))
+                     cu/without-nils)
+        metadata (file-db/file-metadata (du/entity db id new-file))]
+
+    ;; Check if any other file in this task already has the same metadata
+    ;; that this file would have after applying the tx
+    (if (some #(and (not= % id)
+                      (= metadata
```
Comment (outamaa) [inline app/backend/src/clj/teet/file/file_tx.clj:91]: indentation

File: app/backend/test/teet/file/file_commands_test.clj [hunk 3]
```diff
@@ -87,3 +96,62 @@
...
+     (thrown-with-msg?
+      Exception #"Two files with same metadata"
+      (tu/local-command :file/modify
+                        {:db/id (:db/id (tu/get-data :second-file))
+                         :file/name "first file.png"
+                         :file/sequence-number 666
+                         :file/document-group (d/pull (tu/db)
+                                                      [:db/id :db/ident]
+                                                      :file.document-group/general)
+                         :file/part nil})))))
```
Comment (outamaa) [inline app/backend/test/teet/file/file_commands_test.clj:157]: Good tests!

**Review by outamaa — APPROVED** [review]

File: app/backend/src/clj/teet/file/file_tx.clj (hunk 1 above)
Comment (tatut) [reply app/backend/src/clj/teet/file/file_tx.clj:84]: The map value is what we pull/select of the linked entity, but it isn't valid value for a `ref` attribute. That's why we need to just use the `:db/id`

**Pull request merged**